integration/spark-datasource/src/test/scala/org/apache/spark/sql/carbondata/datasource/SparkCarbonDataSourceBinaryTest.scala

```diff
@@ -0,0 +1,304 @@
+                   | options('SORT_COLUMNS'='image')
+       """.stripMargin)
+            sql("SELECT COUNT(*) FROM binaryTable").show()
+        }
+        assert(exception.getCause.getMessage.contains("sort columns not supported for array, struct, map, double, float, decimal, varchar, binary"))
+    }
+
+    test("Don't support long_string_columns for binary") {
+        import spark._
+        sql("DROP TABLE IF EXISTS binaryTable")
+        val exception = intercept[Exception] {
+            sql(
+                s"""
+                   | CREATE TABLE binaryTable (
+                   |    id DOUBLE,
+                   |    label BOOLEAN,
+                   |    name STRING,
+                   |    image BINARY,
+                   |    autoLabel BOOLEAN)
+                   | using carbon
+                   | options('long_string_columns'='image')
+       """.stripMargin)
+            sql("SELECT COUNT(*) FROM binaryTable").show()
+        }
+        assert(exception.getCause.getMessage.contains("long string column : image is not supported for data type: BINARY"))
+    }
+
+    test("Don't support load for datasource") {
+        sql("DROP TABLE IF EXISTS binaryCarbon")
+        if (SparkUtil.isSparkVersionXandAbove("2.2")) {
+            sql(
+                s"""
+                   | CREATE TABLE binaryCarbon(
+                   |    binaryId INT,
+                   |    binaryName STRING,
+                   |    binary BINARY,
+                   |    labelName STRING,
+                   |    labelContent STRING
+                   |) USING CARBON  """.stripMargin)
+
+            val exception = intercept[Exception] {
+                sql(s"load data local inpath '$writerPath' into table binaryCarbon")
+            }
+            assert(exception.getMessage.contains("LOAD DATA is not supported for datasource tables"))
+        }
+        sql("DROP TABLE IF EXISTS binaryCarbon")
+    }
+
+    //TODO: support insert into
+    test("Test partition") {
+        sql("DROP TABLE IF EXISTS binaryCarbon")
+        sql("DROP TABLE IF EXISTS binaryCarbon3")
+        if (SparkUtil.isSparkVersionXandAbove("2.2")) {
+            sql(s"CREATE TABLE binaryCarbon USING CARBON LOCATION '$writerPath'")
+            sql(
+                s"""
+                   | CREATE TABLE binaryCarbon3(
+                   |    binaryId INT,
+                   |    binaryName STRING,
+                   |    binary BINARY,
+                   |    labelName STRING,
+                   |    labelContent STRING
+                   |) USING CARBON partitioned by (binary) """.stripMargin)
+            sql("select binaryId,binaryName,binary,labelName,labelContent from binaryCarbon where binaryId=0").show()
+            try {
+                // TODO: query should show string, not object
+                sql("insert into binaryCarbon3 select binaryId,binaryName,binary,labelName,labelContent from binaryCarbon where binaryId=0 ")
+                assert(false)
```

Comment:
Not sure why this testcase should fail. First add success partition testcase.

Reply:
I think it because select binary will return byte[], not string, insert into doesn't support byte[]

Reply:
insert should be supported

Reply:
yes

Reply:
We shouldn't support insert into binary partition table.